### PR TITLE
GEODE-4993: Store GatewaySender connection stats

### PR DESCRIPTION
**Pull request by alb3rtobr**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [n/a] Have you written or updated unit tests to verify your changes?

- [n/a] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

**Comment by alb3rtobr** [comment]
> Thanks for the contribution!.
> Can you please add some tests to verify your changes?.
> Also, it looks like you're introducing a new (undocumented) statistic into the product, shouldn't you change the [statistics documentation](https://github.com/apache/geode/blob/develop/geode-docs/reference/statistics_list.html.md.erb) as part of this PR as well?.

I completely forgot to check documentation, thanks for pointing that out. I have added a new commit with the changes.

Regarding tests, what should I test? Check if the stats are in the gfs file?

File: geode-core/src/main/java/org/apache/geode/cache/client/internal/ConnectionStats.java [hunk 1]
```diff
@@ -1722,13 +1683,19 @@ private static long getStatTime() {
...
     this.poolStats = poolStats;
   }
 
+  public ConnectionStats(StatisticsFactory factory, String prefix, String name,
+      PoolStats poolStats) {
+    this.stats = factory.createAtomicStatistics(type, prefix + "Stats-" + name);
+    this.sendStats = factory.createAtomicStatistics(sendType, prefix + "SendStats-" + name);
+    this.poolStats = poolStats;
+  }
+
```
[the remark this answers is not in the document]
Comment (alb3rtobr) [reply geode-core/src/main/java/org/apache/geode/cache/client/internal/ConnectionStats.java]: Done!

**Comment by boglesby** [comment]
There are a few existing *StatsTest junit tests (like CachePerfStatsTest). You might add a similar one for ClientStatsTest.

You also might want to take an existing wan dunit test and modify it to test that the ClientStats stats are set properly.

There are a lot of stats in there, but these especially are useful:

- connections
- disconnects
- gatewayBatchs
- pings

You might also want to verify the stats instance is set to the correct type.

The wan dunit tests extend from WANTestBase. One that deals with stats is called ParallelWANStatsDUnitTest. Unfortunately, these tests are pretty complicated. Hopefully, you can just hook into one that is already there.

Here is a method to get the ConnectionStats for a GatewaySender:
```
private void getGatewaySenderConnectionStats(String gatewaySenderId) {
  AbstractGatewaySender sender = (AbstractGatewaySender) CacheFactory.getAnyInstance().getGatewaySender(gatewaySenderId);
  if (sender != null) {
    int totalConnections = 0;
    for (ConnectionStats connectionStats : sender.getProxy().getEndpointManager().getAllStats().values()) {
      totalConnections += connectionStats.getConnections();
    }
    System.out.println("Total connections: " + totalConnections);
  }
}
```

**Comment by mhansonp** [comment]
Hi, I noticed this has not moved in a bit. Do you have any questions? Is there anything someone might be able to assist with? Otherwise, thank you for your efforts.

**Comment by alb3rtobr** [comment]
@mhansonp I could not work on this PR last week, this is why its still on the same state. Thanks!

**Comment by mhansonp** [comment]
No rush, I was just trying to make sure you had all the help you need. Thanks!

**Comment by alb3rtobr** [comment]
@jujoramos @boglesby ConnectionStatsTest already exists, but I have started reviewing the existing tests. I will be uploading the changes in separated commits, I think its better for review.

**Comment by alb3rtobr** [comment]
@jujoramos @boglesby I have added a test to verify the stats are created when a gateway sender is configured. Do you think all these test are enough for this PR?

**Comment by kirklund** [comment]
@jujoramos Do you have time to review the latest update to this PR?

**Comment by mhansonp** [comment]
@jhuynh1  @boglesby Based on the fact that both reviewers approved this change, I am going to just merge it.

**Pull request merged**